src/routes/common/auth/basic-auth.guard.ts

```diff
@@ -0,0 +1,5 @@
+import { Injectable } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+
+@Injectable()
+export class BasicAuthGuard extends AuthGuard('basic-auth') {}
```

Comment:
Have you looked into the `CanActivate` guard?

Maybe this ends up boiling down to the discussion of Authorisation vs Authentication.

It seems that this implementation tries to use an authentication framework (`passport`) to protect the endpoints. However maybe just a simple authorisation guard might be enough? (since we do not have a concept of a user)

Example:

```typescript
@Injectable()
export class BasicAuthGuard implements CanActivate {
  constructor(
    @Inject(IConfigurationService)
    private readonly configurationService: IConfigurationService,
  ) {}

  canActivate(context: ExecutionContext): boolean | Promise<boolean> {
    const request = context.switchToHttp().getRequest();
    const token = this.configurationService.getOrThrow('auth.token');
    return request.headers['Authorization'] === `Basic ${token}`;
  }
}
```

```typescript
@UseGuards(BasicAuthGuard)
@Post('/chains/:chainId/hooks/events')
@HttpCode(200)
async postEvent(
  ...
)
```

Reply:
Good point, I also prefer to keep the number of dependencies at a minimum. I didn't explore `CanActivate` since it's more oriented to Authorization as you said, but maybe it makes sense in this use case. I'll try this solution and share the outcome, thanks.